pkg/atomicwriter/atomic_writer_test.go

```diff
@@ -150,47 +283,41 @@ func TestLogsErrors(t *testing.T) {
 		{
 			name: "unable to remove or truncate temporary file",
 			path: "test_file.txt",
-			osFuncs: osFuncs{
-				remove: func(name string) error {
-					return os.ErrPermission
-				},
-				rename: func(oldpath, newpath string) error {
-					return os.ErrPermission
-				},
-				truncate: func(name string, size int64) error {
-					return os.ErrPermission
-				},
-				chmod: os.Chmod,
+			injectErrs: injectErrors{
+				errOnRename:   true,
+				errOnRemove:   true,
+				errOnTruncate: true,
 			},
-			assert: func(buf *bytes.Buffer, wc io.WriteCloser, t *testing.T, err error) {
+			assert: func(t *testing.T, buf *bytes.Buffer, wc io.WriteCloser, err error) {
 				assert.Error(t, err)
 
 				assert.Contains(t, buf.String(), "Could not delete temporary file")
 				assert.Contains(t, buf.String(), "File may be left on disk")
 			},
 		},
 		{
-			name: "unable to chmod",
-			path: "test_file.txt",
-			osFuncs: osFuncs{
-				remove:   os.RemoveAll,
-				rename:   os.Rename,
-				truncate: os.Truncate,
-				chmod: func(name string, mode os.FileMode) error {
-					return os.ErrPermission
-				},
-			},
-			assert: func(buf *bytes.Buffer, wc io.WriteCloser, t *testing.T, err error) {
+			name:       "unable to chmod",
+			path:       "test_file.txt",
+			injectErrs: injectErrors{errOnChmod: true},
+			assert: func(t *testing.T, buf *bytes.Buffer, wc io.WriteCloser, err error) {
 				assert.NoError(t, err)
 				assert.Contains(t, buf.String(), "Could not set permissions on temporary file")
 
 				// Check that the file was still renamed
 				writer, ok := wc.(*atomicWriter)
 				assert.True(t, ok)
-				assert.NoFileExists(t, writer.tempFile.Name())
```

Comment:
I wonder if we should actually add this line to more of the failure tests. Thoughts?

Reply:
Yeah, we can add checks that the temp file does not exist, but we'll have to capture/record the temp file name just after we call writer.Write(), since the `tempFile` pointer in the `atomicWriter` struct is now getting nil'd out whenever the temp file has in fact been deleted.

I'll add the capture of the temp file name and the checks that the temp file has been deleted where it makes sense.

I also noticed that we're not checking for `IsNotExist(err)` when the call to `os.remove()` fails, so I'll add that in.

Also, when `os.remove()` fails, and then we do an `os.truncate()`, if the truncate ever returns `os.ErrNotExist` (granted, this should theoretically never happen, but for the purposes of safe coding...), we should just set `writer.tempFile = nil` and return without any error messages.